test/WebSites/RazorPageExecutionInstrumentationWebSite/Startup.cs

```diff
@@ -22,24 +20,28 @@ public void ConfigureServices(IServiceCollection services)
 
             // Add MVC services to the services container.
             services.AddMvc();
-
-            // Make instrumentation data available in views.
-            services.AddScoped<TestPageExecutionListenerFeature, TestPageExecutionListenerFeature>();
-            services.AddScoped<IHoldInstrumentationData>(
-                provider => provider.GetRequiredService<TestPageExecutionListenerFeature>().Holder);
         }
 
         public void Configure(IApplicationBuilder app)
         {
+            var listener = new RazorPageDiagnosticListener();
+            var diagnosticSource = app.ApplicationServices.GetRequiredService<DiagnosticListener>();
+            diagnosticSource.SubscribeWithAdapter(listener);
```

Comment:
This setup looks suspect because the only registered `DiagnosticListener` is a singleton. That implies this site can be used once and only once, undermining the text fixtures we use.

Also odd because the `diagnosticSource` has type `DiagnosticListener`.